apps/dashboard/src/main/java/com/akto/action/user/GithubSsoAction.java

```diff
@@ -45,8 +65,174 @@ public String deleteGithubSso() {
+        try {
+            Map<String, String> metaData = testingRunResultSummary.getMetadata();
+            String repository = metaData.get("repository");
+            String commitSHA = metaData.get("commit_sha_head");
+            jwtToken = JWT.createJWT(githubAppId,privateKey, 10 * 60 * 1000);
+            GitHub gitHub = new GitHubBuilder().withJwtToken(jwtToken).build();
+            GHApp ghApp = gitHub.getApp();
+
+            //Getting appInstallations
+            List<GHAppInstallation> appInstallations = ghApp.listInstallations().toList();
+            if (appInstallations.isEmpty()) {
+                addActionError("Github app was not installed");
+                return ERROR.toUpperCase();
+            }
+
+            GHAppInstallation appInstallation = appInstallations.get(0);
+            GHAppCreateTokenBuilder builder = appInstallation.createToken();
+            GHAppInstallationToken token = builder.create();
+            GitHub githubAccount =  new GitHubBuilder().withAppInstallationToken(token.getToken())
+                    .build();
+
+            GHRepository ghRepository = githubAccount.getRepository(repository);
+            if (ghRepository == null) {
+                addActionError("Github app doesn't have access to repository");
+                return ERROR.toUpperCase();
+            }
+            GHCheckRunBuilder ghCheckRunBuilder = ghRepository.createCheckRun("Akto Security Checks", commitSHA);
+            ghCheckRunBuilder.withStatus(GHCheckRun.Status.IN_PROGRESS)
+                    .withStartedAt(new Date())
+                    .add(new GHCheckRunBuilder.Output("Akto CI/CD test running", ""))
+                    .create();
+        } catch (Exception e) {
+            addActionError("Error while publishing github checks");
+            loggerMaker.errorAndAddToDb("Github checks error : " + e.getMessage(), LoggerMaker.LogDb.DASHBOARD);
+            return ERROR.toUpperCase();
+        }
+
+
+        return SUCCESS.toUpperCase();
+    }
+    public String publishGithubComments() {
+        AccountSettings accountSettings = AccountSettingsDao.instance.findOne(generateFilter());
+        String privateKey = accountSettings.getGithubAppSecretKey();
+        String githubAppId = accountSettings.getGithubAppId();
+        TestingRunResultSummary testingRunResultSummary = TestingRunResultSummariesDao.instance.findOne(Filters.eq(ID,new ObjectId(testingRunSummaryHexId)));
+        try {
+            Map<String, String> metaData = testingRunResultSummary.getMetadata();
+            String repository = metaData.get("repository");
+            String pullRequestId = metaData.get("pull_request_id");
+            String commitSHA = metaData.get("commit_sha_head");
+            boolean isCompleted = testingRunResultSummary.getState() == TestingRun.State.COMPLETED;
+            StringBuilder messageStringBuilder = new StringBuilder();
```

Comment:
Is this repetitive code between `publishGithubComments` and `updateGithubStatus` ?